python_bindings/tests/end-to-python-end-test/all_tiles_test.py

```diff
@@ -0,0 +1,1460 @@
+           ..|.........|...\ /..\ /..\ /.
+2          --6----8----A....C...|D|..|E|.
+           ..|.../ \../ \../ \../ \../ \.
+           ..|..|   || @ ||   || ! || @ |
+           ..|..|   |                   |
+           ..|...\ /        *          /.
+1          --5....F !       L    M    G..
+           ..|.....\       /.\  /|\  /...
+           ..|......-     -...--.|.--..@.
+      -......|.......|   |.....
+       \.....|........\ /..[@].
+0     * H....4----3----0---[2].
+       ! \.................[ ].
+          |..@.................
+                     .........--...!
+                     .[!]..../  \... 
+-1                   .[1]...7    9--     
+                     .[ ]....\  /|..
+                     .........--.|..
+
+"""
+
+
+def check_turn_N(engine: GameEngine, game, game_id) -> (int, SerializedGame):
+    turn_params = TurnParams(
+        pos=Position(x=0, y=4),
+        tile=game.current_tile,
+        meepleType=MeepleType.NoneMeeple,
+        side=Side.NoSide,
+        featureType=FeatureType.NoneType,
+    )
+
+    game_id, game = make_turn(engine, game, game_id, turn_params)
+    check_points(game, [11, 8])
+
+    return game_id, game
+
+
+"""
+both players score: 13*2 + 3*2 = 26+6=32 (huge city)
+    14 tile city
+    3 shield in a city
+
+       -3   -2   -1    0    1    2    3
+
+
+                     -.|.-
+                      \|/ 
+4                    * N    
+                          
+                                                      
+                     |   |..@.--...--.|..    
+                     .\ /...|/  \./  \|..  
+3                    --B----I    K    J-! 
+                     ..|.../         * \.
+                     !.|..|             |
+           ..|.........|..|   ||   || * |
+           ..|.........|...\ /..\ /..\ /.
+2          --6----8----A....C...|D|..|E|.
+           ..|.../ \../ \../ \../ \../ \.
+           ..|..|   || @ ||   || ! || @ |
+           ..|..|   |                   |
+           ..|...\ /        *          /.
+1          --5....F !  O    L    M    G..
+           ..|.....\       /.\  /|\  /...
+           ..|......-     -...--.|.--..@.
+      -......|.......|   |.....
+       \.....|........\ /..[@].
+0    *  H....4----3----0---[2].
+       ! \.................[ ].
+          |..@.................
+                     .........--...!
+                     .[!]..../  \... 
+-1                   .[1]...7    9--     
+                     .[ ]....\  /|..
+                     .........--.|..
+               
+"""
+
+
+def check_turn_O(engine: GameEngine, game, game_id) -> (int, SerializedGame):
+    turn_params = TurnParams(
+        pos=Position(x=0, y=1),
+        tile=game.current_tile,
+        meepleType=MeepleType.NoneMeeple,
+        side=Side.NoSide,
+        featureType=FeatureType.NoneType,
+    )
+
+    game_id, game = make_turn(
+        engine, game, game_id, turn_params, True, {1: 11 + 34 + 11, 2: 8 + 34 + 22}
```

Comment:
I'm probably misunderstanding this, but why do you need to tell make_turn it's the last turn and what the scores are? Shouldn't the engine already know this? If it's possible I'd rather want this function to automatically detect that the game has ended and final tile was placed.

Reply:
removed param

Reply:
I think the `final_scores` param could also be removed. In all cases you are calling `check_points()` after `make_turn()` anyway, so `make_turn()` doesn't need to itself check the scores.

(This change would also make `check_final_points()` obsolete - I think it does the same thing as `check_points()` so it could be removed, but I haven't looked into it too deeply)

Reply:
Actually, not really.  SerializedGame has points directly after the move is made, but it doesn't take into account the finalizasion process. Final score is in MakeTurnResponse, which i don't return with the make_turn function, because it is nowhere used apart from final scoring. You can see that check_points call has much less points. Technically make_turn and check_points could be merged, but i wanted to add some distinction, and similarity to go side tests.

Reply:
Ok, I understand. In that case, I don't have any further comments, approved

I created a new issue (#139) because I don't think this is the desired behaviour, you can take a look at it to make sure I understood everything correctly